depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp

```diff
@@ -91,22 +99,29 @@ class DeviceROS : public dai::DeviceBase {
     template <class MsgType>
     auto generate_pub_lambda(ros::NodeHandle& nh, std::string name, std::size_t q_size) {
         auto conn = this->getConnection();
-        const auto pub_lambda = [this, pub = nh.advertise<MsgType>(name, q_size),
-                                        // no writing happens, so 1 is sufficient
-                                        stream = dai::XLinkStream{*conn, name, 1}]() {
+        _pub_map[name] = nh.advertise<MsgType>(name, q_size); // no writing happens, so 1 is sufficient
+        _streams[name] = std::make_unique<dai::XLinkStream>(*conn, name, 1);
+        const auto pub_lambda = [this, name]() {
             Guard guard([] { ROS_ERROR("Communication failed: Device error or misconfiguration."); });
 
             while (this->_running) {
                 // block till data is read
-                PacketReader reader{stream};
-                auto data = reader.getData();
-                // @TODO convert data to ROS message type here, somehow
+                PacketReader reader{*_streams[name]};
```

Comment:
The capture of stream and publisher was done to eliminate the need for a lookup in a map